### PR TITLE
protectli vault ehl devicetree

### PR DESCRIPTION
**Pull request by mkopec**
Signed-off-by: Michał Kopeć <michal.kopec@3mdeb.com>

### CODE REVIEW

File: src/mainboard/protectli/vault_ehl/devicetree.cb [hunk 1]
```diff
@@ -24,86 +22,175 @@ chip soc/intel/elkhartlake
...
+		device pci 1d.4 off	end # Intel PSE DMA1
+		device pci 1d.5 off	end # Intel PSE DMA2
+		device pci 1d.6 off	end # Intel PSE PWM
+		device pci 1d.7 off	end # Intel PSE ADC
+
+		device pci 1e.0 on	end # UART0
+		device pci 1e.1 on	end # UART1
+		device pci 1e.2 on	end # GSPI0
+		device pci 1e.3 on	end # GSPI1
+		device pci 1e.4 on	end # PCH Time-Sensitive Networking GbE
```
Comment (miczyg1) [inline src/mainboard/protectli/vault_ehl/devicetree.cb:168]: These should be disabled?

File: src/mainboard/protectli/vault_ehl/devicetree.cb [hunk 2]
```diff
@@ -24,86 +22,175 @@ chip soc/intel/elkhartlake
...
+			end
+			chip drivers/pc80/tpm
+				device pnp 0c31.0 on end
 			end
 		end
-		device pci 1f.1 on  end	# - SMBUS
+		device pci 1f.1 on	end # P2SB
+		device pci 1f.2 hidden	end # Power Management Controller
+		device pci 1f.3 on	end # Intel cAVS/HDA
+		device pci 1f.4 on  end	# - SMBUS
```
Comment (miczyg1) [inline src/mainboard/protectli/vault_ehl/devicetree.cb:192]: Indentation

File: src/mainboard/protectli/vault_ehl/devicetree.cb [hunk 3]
```diff
@@ -1,15 +1,13 @@
...
 
-	#register "dptf_enable" = "1"
 	#register "enable_vtd" = "1"
 
 	register "power_limits_config" = "{
 		.tdp_pl1_override = 10,
 		.tdp_pl2_override = 15,
 	}"
 
-
```
Comment (miczyg1) [inline src/mainboard/protectli/vault_ehl/devicetree.cb:12]: SaGv = SaGv_Enabled ?
eist_enable = 1 ?

File: src/mainboard/protectli/vault_ehl/devicetree.cb [hunk 4]
```diff
@@ -24,86 +22,175 @@ chip soc/intel/elkhartlake
...
 	register "usb2_ports[9]" = "USB2_PORT_EMPTY"
 
-	register "usb3_ports[0]" = "USB3_PORT_DEFAULT(OC_SKIP)"	# USB3.0 Type A
-	register "usb3_ports[1]" = "USB3_PORT_DEFAULT(OC_SKIP)"	# USB3.0 Type A
-	register "usb3_ports[2]" = "USB3_PORT_DEFAULT(OC_SKIP)"	# USB3.0 Type C
+	# USB 3.x ports
+	register "usb3_ports[0]" = "USB3_PORT_DEFAULT(OC_SKIP)"	# USB Type-A Upper
+	register "usb3_ports[1]" = "USB3_PORT_DEFAULT(OC_SKIP)"	# USB Type-A Lower
+	register "usb3_ports[2]" = "USB3_PORT_DEFAULT(OC_SKIP)"	# USB Type-C Muxed
 	register "usb3_ports[3]" = "USB3_PORT_EMPTY"
```
Comment (miczyg1) [inline src/mainboard/protectli/vault_ehl/devicetree.cb:42]: usb2_wake_enable_bitmap ?
usb3_wake_enable_bitmap ?

We should set the bits for at least all USB type A and Type C ports

File: src/mainboard/protectli/vault_ehl/devicetree.cb [hunk 5]
```diff
@@ -24,86 +22,175 @@ chip soc/intel/elkhartlake
...
+	register "PcieClkSrcClkReq[4]" = "PCIE_CLK_NOTUSED"
+	register "PcieClkSrcClkReq[5]" = "PCIE_CLK_NOTUSED"
+
+	# Storage (SATA/SDCARD/EMMC) related UPDs
+	register "SataSalpSupport" = "1"
+	register "SataPortsEnable[0]" = "1" # Header
+	register "SataPortsEnable[1]" = "1" # M.2 2280
+	register "SataPortsDevSlp[0]" = "0"
+	register "SataPortsDevSlp[1]" = "1"
+
```
Comment (miczyg1) [inline src/mainboard/protectli/vault_ehl/devicetree.cb:70]: PchHdaAudioLinkHdaEnable = 1 ?
PchHdaSdiEnable[0] = 1 ?
ScsEmmcHs400Enabled = 1 ?
SkipCpuReplacementCheck = 1 ?

DdiPortAConfig ? 
DdiPortBConfig ?
DdiPort*Hpd ?
DdiPort*Ddc ?

**Review by miczyg1 — CHANGES_REQUESTED** [review]

File: src/mainboard/protectli/vault_ehl/devicetree.cb [hunk 6]
```diff
@@ -24,86 +22,175 @@ chip soc/intel/elkhartlake
...
+		device pci 13.0 off	end # Intel PSE GSPI0
+		device pci 13.1 off	end # Intel PSE GSPI1
+		device pci 13.2 off	end # Intel PSE GSPI2
+		device pci 13.3 off	end # Intel PSE GSPI3
+		device pci 13.4 off	end # Intel PSE GPIO0
+		device pci 13.5 off	end # Intel PSE GPIO1
+
+		device pci 14.0 on	end # USB3.1 xHCI
+		device pci 14.1 off	end # USB3.1 xDCI (OTG)
+		device pci 14.2 on	end # USB Type-C UCSI?
```
[the remark this answers is not in the document]
Comment (mkopec) [reply src/mainboard/protectli/vault_ehl/devicetree.cb:109]: Present in stock firmware logs, device `00:14.2 RAM memory [0500]: Intel Corporation Device [8086:4b7f] (rev 11)`.

File: src/mainboard/protectli/vault_ehl/devicetree.cb [hunk 7]
```diff
@@ -24,86 +22,175 @@ chip soc/intel/elkhartlake
...
+		device pci 11.4 off	end # Intel PSE UART4
+		device pci 11.5 off	end # Intel PSE UART5
+		device pci 11.6 off	end # Intel PSE IS20
+		device pci 11.7 off	end # Intel PSE IS21
+
+		device pci 12.0 on	end # GSPI2
+		device pci 12.3 on	end # Management Engine UMA Access
+		device pci 12.4 on	end # Management Engine PTT DMA Controller
+		device pci 12.5 off	end # UFS0
+		device pci 12.7 off	end # UFS1
```
[the remark this answers is not in the document]
Comment (mkopec) [reply src/mainboard/protectli/vault_ehl/devicetree.cb:98]: Copied from ehlcrb upon which this devicetree is based

File: src/mainboard/protectli/vault_ehl/devicetree.cb (hunk 6 above)
[the remark this answers is not in the document]
Comment (miczyg1) [reply src/mainboard/protectli/vault_ehl/devicetree.cb:109]: If anything this is shared_ram, also present in TGL.

File: src/mainboard/protectli/vault_ehl/devicetree.cb [hunk 8]
```diff
@@ -24,86 +22,175 @@ chip soc/intel/elkhartlake
...
+		device pci 1b.5 off	end # Intel PSE I2C5
+		device pci 1b.6 off	end # Intel PSE I2C6
+
+		device pci 1c.0 on	end # RP1 (pcie0 single VC)
+		device pci 1c.1 on	end # RP2 (pcie0 single VC)
+		device pci 1c.2 on	end # RP3 (pcie0 single VC)
+		device pci 1c.3 off	end # RP4 (pcie0 single VC)
+		device pci 1c.4 on	end # RP5 (pcie1 multi VC)
+		device pci 1c.5 off	end # RP6 (pcie2 multi VC)
+		device pci 1c.6 off	end # RP7 (pcie3 multi VC)
```
[the remark this answers is not in the document]
Comment (mkopec) [reply src/mainboard/protectli/vault_ehl/devicetree.cb]: Fixed

File: src/mainboard/protectli/vault_ehl/devicetree.cb (hunk 5 above)
Comment (mkopec) [reply src/mainboard/protectli/vault_ehl/devicetree.cb:73]: Added DDC / HPD params for DdiPortA and DdiPortC, the rest aren't wired up in soc code

File: src/mainboard/protectli/vault_ehl/devicetree.cb (hunk 4 above)
Comment (mkopec) [reply src/mainboard/protectli/vault_ehl/devicetree.cb:45]: These are only set by skylake and cannonlake boards, do we need them?

File: src/mainboard/protectli/vault_ehl/devicetree.cb (hunk 5 above)
Comment (miczyg1) [reply src/mainboard/protectli/vault_ehl/devicetree.cb:73]: Then we have to wire it up.

File: src/mainboard/protectli/vault_ehl/devicetree.cb (hunk 8 above)
[the remark this answers is not in the document]
Comment (miczyg1) [reply src/mainboard/protectli/vault_ehl/devicetree.cb]: But still the code doesn't match settings from FSP. YOu have off-by-one error

File: src/mainboard/protectli/vault_ehl/devicetree.cb (hunk 2 above)
Comment (miczyg1) [reply src/mainboard/protectli/vault_ehl/devicetree.cb]: SMBus has wrong indentation still

File: src/mainboard/protectli/vault_ehl/devicetree.cb [hunk 9]
```diff
@@ -24,86 +25,188 @@ chip soc/intel/elkhartlake
...
-#			chip drivers/pc80/tpm
-#				device pnp 0c31.0 on end
-#			end
-			chip superio/ite/it8613e 
-				device pnp 2e.0 off end 
-				device pnp 2e.1 on      # COM 1 
-					io 0x60 = 0x3f8 
-					irq 0x70 = 4 
-				end 
+		device pci 00.0 on	end # Host Bridge
```
Comment (miczyg1) [inline src/mainboard/protectli/vault_ehl/devicetree.cb:91]:
```suggestion
		device pci 00.0 on  end # Host Bridge
```

Two spaces for on and one space for off devices. Hidden also one space.

File: src/mainboard/protectli/vault_ehl/devicetree.cb (hunk 8 above)
[the remark this answers is not in the document]
Comment (mkopec) [reply src/mainboard/protectli/vault_ehl/devicetree.cb]: fixed as per suggestion

File: src/mainboard/protectli/vault_ehl/devicetree.cb (hunk 5 above)
Comment (mkopec) [reply src/mainboard/protectli/vault_ehl/devicetree.cb:73]: done

File: src/mainboard/protectli/vault_ehl/devicetree.cb (hunk 2 above)
Comment (mkopec) [reply src/mainboard/protectli/vault_ehl/devicetree.cb]: fixed

File: src/mainboard/protectli/vault_ehl/devicetree.cb (hunk 8 above)
[the remark this answers is not in the document]
Comment (mkopec) [reply src/mainboard/protectli/vault_ehl/devicetree.cb]: How so? I wasn't able to get all 4 NICs working with your suggestion.

[the remark this answers is not in the document]
Comment (miczyg1) [reply src/mainboard/protectli/vault_ehl/devicetree.cb]: E.g. `device pci 1c.0 on	end # RP1 (pcie0 single VC)` should match `register "PcieRpEnable[0]" = "1"` but we do nto enable `PcieRpEnable[0]` in FSP.

File: src/mainboard/protectli/vault_ehl/devicetree.cb (hunk 5 above)
Comment (miczyg1) [reply src/mainboard/protectli/vault_ehl/devicetree.cb:73]: Good. Except that DDIPortC has no HDP and DDC connected on the schematics (this is the USB-C DP)

File: src/mainboard/protectli/vault_ehl/devicetree.cb (hunk 8 above)
[the remark this answers is not in the document]
Comment (miczyg1) [reply src/mainboard/protectli/vault_ehl/devicetree.cb]: Now all good.

File: src/mainboard/protectli/vault_ehl/devicetree.cb (hunk 5 above)
Comment (mkopec) [reply src/mainboard/protectli/vault_ehl/devicetree.cb:73]: Removed HPD/DDC config for Port C.

**Review by miczyg1 — APPROVED** [review]

**Pull request merged**